sktime/tests/test_mapa.py

```diff
@@ -0,0 +1,213 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from sktime.forecasting.mapa import MAPAForecaster
+
+
+@pytest.fixture
+def sample_data():
+    """Create sample time series data for testing.
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame containing a sample time series with a sinusoidal pattern
+        and some added noise.
+    """
+    np.random.seed(42)
+    dates = pd.date_range(start="2020-01-01", periods=24, freq="M")
+    data = np.sin(np.linspace(0, 4 * np.pi, 24)) * 10 + np.random.normal(0, 1, 24) + 20
+    return pd.DataFrame(data, index=dates, columns=["value"])
+
+
+def test_decompose():
+    """Test the `_decompose` method of the MAPAForecaster class.
+
+    Verifies the following:
+    - The method correctly handles decomposition for different aggregation levels.
+    - The returned decomposed time series matches the input data.
+    - The seasonal period and seasonal flag are calculated correctly.
+    """
+
+    test_params = MAPAForecaster.get_test_params()
+
+    for params in test_params:
+        params["aggregation_levels"] = params.get("aggregation_levels", [1, 2, 4])
+        forecaster = MAPAForecaster(**params)
+        y = pd.DataFrame({"value": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]})
+
+        for level in params["aggregation_levels"]:
+            y_decomp, seasonal_enabled, seasonal_period = forecaster._decompose(
+                y, level
+            )
+
+            # Check if returned data matches input
+            assert isinstance(y_decomp, pd.DataFrame)
+            assert y_decomp.equals(y)
```

Comment:
maybe a silly question - if `y_decomp` always matches `y`, why is it an output? We could just use `y`.